app/controllers/signs_controller.rb

```diff
@@ -79,7 +79,8 @@ def check_contribution_limit!
   end
 
   def signs
-    policy_scope(Sign).for_cards.order(word: :asc)
+    sort_by = Sign::SORT_OPTIONS[params[:sort_by].try(:to_sym)] || Sign::SORT_OPTIONS[:alphabetical]
```

Comment:
It's a good idea to avoid using `try`. What you're attempting to do here is to handle the value being nil, but `try` will swallow _all_ exceptionions - even those you didn't mean to handle. Instead, use conditional acccess `&.`. This handles nil values, but won't swallow exceptions in the course of executing `to_sym` if the value is present.

Reply:
I'm also not wild about the idea of establishing a dependency between the model and the controller here, especially when the model doesn't do anything further with applying the order clause.

Have you considered defining `SORT_OPTIONS` as a const in the controller instead? This would still provide pretty much the behaviour you're looking for, but will keep those values in the place they are used.

Reply:
I think the main thing that is the sort should be applied in the same location the options are defined. A service object would be ideal, since this allows for growth and reusability, but could be overkill here. The controller is a good middle ground, especially since this ordering is only used on this page, but if we add additional sort options, or make the existing ones more complex, we'll have to move to a service object quite quickly. We could also put the order application method in the model, but IMO this is the worst of both worlds - the logic would be separated from where it is used, and also could not grow much in scope or complexity without affecting the rest of the model.